dense-analysis/ale
# Solidity solc linter compatible with 0.6/0.7/0.8

- Add override executable for solc command (for truffle/hardhat binaries is really useful)
- Updating `solc` linter to be compatible with the Error/Warning output for compilers 0.6, 0.7 and 0.8.
- Add a new option to use the old parser (would be great if @karlb check this to be really sure =) )

Manually tested with:
- `let g:ale_solidity_solc_executable='/solc-linux-amd64-v0.6.6+commit.6c089d02`
- `let g:ale_solidity_solc_executable='/solc-linux-amd64-v0.7.1+commit.f4a555be`
- `let g:ale_solidity_solc_executable='/solc-linux-amd64-v0.8.4+commit.c7e474f2` (my current version)
All of them has the same output.

The `./run-tests` command result:
```
Docker run image denseanalysis/ale:7e12f9057ee4621436f0606a90a04556 ready
Starting Vim: neovim-v0.2.0...
Starting Vim: neovim-v0.4.4...
Starting Vim: vim-v8.0.0027...
Starting Vim: vim-v8.2.2401...
Starting Vint...
Starting Custom checks...

Success/Total: 2583/2583 (assertions: 3884/3884)
Lints: Nothing to report
Python code: 5 test OK
```

If there's anything to improve/change or re check let me know.
